cogs/wordfilter/wordfilter.py

```diff
@@ -272,69 +272,56 @@ async def _channel(self, ctx):
     @_channel.command(name="add")
     @commands.guild_only()
     @checks.mod_or_permissions(manage_messages=True)
-    async def _channelAdd(self, ctx, channelName):
+    async def _channelAdd(self, ctx, channel: discord.TextChannel):
         """Add a channel to the allowlist.
 
         All messages in the channel will not be filtered.
 
         Parameters:
         -----------
-        channelName: str
+        channel: discord.TextChannel
             The channel to add to the allowlist.
         """
-        guildId = ctx.message.guild.id
         channelAllowed = await self.config.guild(ctx.guild).channelAllowed()
 
-        match = re.search(PATTERN_CHANNEL_ID, channelName)
-        if match:  # channel ID
-            channel = discord.utils.get(ctx.message.guild.channels, id=match.group(1))
-            channelName = channel.name
-
-        if channelName not in channelAllowed:
-            channelAllowed.append(channelName)
+        if channel.id not in channelAllowed:
+            channelAllowed.append(channel.id)
             await self.config.guild(ctx.guild).channelAllowed.set(channelAllowed)
             await ctx.send(
                 ":white_check_mark: Word Filter: Channel with name "
-                f"`{channelName}` will not be filtered."
+                f"`{channel.name}` will not be filtered."
             )
         else:
             await ctx.send(
                 ":negative_squared_cross_mark: Word Filter: Channel "
-                f"`{channelName}` is already on the allowlist."
+                f"`{channel.name}` is already on the allowlist."
             )
 
     @_channel.command(name="del", aliases=["delete", "remove", "rm"])
     @commands.guild_only()
     @checks.mod_or_permissions(manage_messages=True)
-    async def _channelRemove(self, ctx, channelName):
+    async def _channelRemove(self, ctx, channel: discord.TextChannel):
         """Remove a channel from the allowlist.
 
         All messages in the removed channel will be subjected to the filter.
 
         Parameters:
         -----------
-        channelName: str
+        channel: discord.TextChannel
             The channel to remove from the allowlist.
         """
-        guildName = ctx.message.guild.name
-
         channelAllowed = await self.config.guild(ctx.guild).channelAllowed()
 
-        match = re.search(PATTERN_CHANNEL_ID, channelName)
-        if match:  # channel ID
-            channel = discord.utils.get(ctx.message.guild.channels, id=match.group(1))
-            channelName = channel.name
-
-        if not channelAllowed or channelName not in channelAllowed:
+        if not channelAllowed or channel.id not in channelAllowed:
```

Comment:
nit: can simplify
```suggestion
        if channel.id not in channelAllowed:
```

Reply:
Actually, here and everywhere else where `channelAllowed` is referenced: can we rename `channelAllowed` to `channelIdsAllowed`? Since this is a breaking change, we should map this directly into a new attribute name.